### PR TITLE
GEODE-4421: Convert Execution factory to value type

### PR DESCRIPTION
**Pull request by igodwin**

### CODE REVIEW

File: clicache/src/Execution.hpp [hunk 1]
```diff
@@ -91,25 +91,25 @@ namespace Apache
         /// <returns>
         /// The managed wrapper object; null if the native pointer is null.
         /// </returns>
-        inline static Execution<TResult>^ Create( std::shared_ptr<native::Execution> nativeptr, IResultCollector<TResult>^ rc )
+        inline static Execution<TResult>^ Create(native::Execution&& nativeExecution, IResultCollector<TResult>^ rc )
         {
-          return __nullptr == nativeptr ? nullptr :
-            gcnew Execution<TResult>( nativeptr, rc );
+            return gcnew Execution<TResult>( std::move(nativeExecution), rc );
```
Comment (pivotal-jbarrett) [inline clicache/src/Execution.hpp]: Looks like some formatting issues here.

File: cppcache/include/geode/ResultCollector.hpp [hunk 2]
```diff
@@ -50,8 +50,8 @@ namespace client {
  *  <br>
  *  <pre>
  * auto rc = FunctionService::onRegion(region)
- *                                      ->withArgs(args)
- *                                      ->withFilter(keySet)
+ *                                      .withArgs(args)
+ *                                      .withFilter(keySet)
  *                                      ->withCollector(new
```
Comment (pivotal-jbarrett) [inline cppcache/include/geode/ResultCollector.hpp]: This line should probably be dot not arrow.

File: cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp [hunk 3]
```diff
@@ -783,66 +780,64 @@ DUNIT_TASK_DEFINITION(WRITER_CLIENT, StepTwo)
...
 
-     //----------------------------------------------------------------------------------------------//
+      //----------------------------------------------------------------------------------------------//
 
-     // FunctionService::onServer(pool)->execute("securityTest",
-     // true)->getResult();
-     // FAIL("Function execution should not have completed successfully");
-     // FunctionService::onRegion(regionPtr)->execute("securityTest",
-     // true)->getResult();
+      // FunctionService::onServer(pool)->execute("securityTest",
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp]: Just delete this commented out code.

File: cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp [hunk 4]
```diff
@@ -919,47 +840,27 @@ DUNIT_TASK_DEFINITION(READER_CLIENT, StepThree)
...
-               checkPtr->value().c_str(), keys[2]);
-       LOG(buf);
-     } else {
-       LOG("checkPtr is nullptr");
-     }
-    }
-    HANDLE_NO_NOT_AUTHORIZED_EXCEPTION
-
-    //auto regPtr0 = getHelper()->getRegion( regionNamesAuth[0] );
-    if (rptr != nullptr) {
```
Comment (igodwin) [inline cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp]: Ripped this, but I don't know if it should be made to actually do something, instead.

File: clicache/src/Execution.hpp (hunk 1 above)
Comment (igodwin) [reply clicache/src/Execution.hpp]: Fixed formatting.

File: cppcache/include/geode/ResultCollector.hpp (hunk 2 above)
Comment (igodwin) [reply cppcache/include/geode/ResultCollector.hpp]: Done.

File: cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp (hunk 3 above)
Comment (igodwin) [reply cppcache/integration-test/testThinClientSecurityAuthorizationMU.cpp]: Done.

**Pull request merged**